File: ackermann.js
let ackermann = function(m, n) {
  if (m === 0) {
    n = n + 1;
    return n;
  } else if (m > 0 && n == 0) {
    return ackermann(m - 1, 1);
  } else if (m > 0 && n > 0) {
    return ackermann(m - 1, ackermann(m, n - 1));
  }
};

console.log(ackermann(4, 2));

File: ackermann.fs
let rec ackermann m n : int =
  match m, n with
  | 0, n -> n + 1
  | m, 0 -> ackermann (m - 1) 1
  | m, n -> ackermann (m - 1) (ackermann m (n - 1))

do
  printfn "%i" (ackermann 0 2)
